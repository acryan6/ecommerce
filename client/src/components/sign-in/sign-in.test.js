import { shallow } from "enzyme";
import React from "react";
import { SignIn } from "./sign-in.component.jsx";
import FormInput from "../form-input/form-input.component";

describe("sign-in component", () => {
  let wrapper;
  let mockEmailSignInStart;
  let mockGoogleSignInStart;

  beforeEach(() => {
    mockEmailSignInStart = jest.fn();
    mockGoogleSignInStart = jest.fn();

    const mockProps = {
      emailSignInStart: mockEmailSignInStart,
      googleSignInStart: mockGoogleSignInStart,
    };

    wrapper = shallow(<SignIn {...mockProps} />);
  });

  it("should render sign-in component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("correctly handles submit", () => {
    const preventDefault = jest.fn();
    wrapper.find("form").simulate("submit", { preventDefault });

    expect(mockEmailSignInStart).toHaveBeenCalled();
  });

  it("calls googleSignInStart", () => {
    wrapper.find("CustomButton").at(1).simulate("click");

    expect(mockGoogleSignInStart).toHaveBeenCalled();
  });
});
