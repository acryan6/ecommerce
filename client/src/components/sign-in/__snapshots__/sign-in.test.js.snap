// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign-in component should render sign-in component 1`] = `ShallowWrapper {}`;
